### PR TITLE
fix: participant details do not update (ACC-107)

### PR DESCRIPTION
**Pull request by przemvs**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/ACC-107" title="ACC-107" target="_blank">BugACC-107</a>  [Web] Participant details do not update
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

_Briefly describe the issue you have solved or implemented with this pull request. If the PR contains multiple issues, use a bullet list._

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: src/script/components/panel/UserActions.tsx [hunk 1]
```diff
@@ -94,6 +94,15 @@ const UserActions: React.FC<UserActionsProps> = ({
   conversationRoleRepository,
   selfUser,
 }) => {
+  useKoSubscribableChildren(user, [
```
Comment (tlebon) [inline src/script/components/panel/UserActions.tsx]: feels like destructuring this and/or not passing in the user prop *to the component* would be more appropriate based on other patterns we are using.

Comment (tlebon) [reply src/script/components/panel/UserActions.tsx]: Screenshot 2022-05-06 at 12 35 04

Comment (przemvs) [reply src/script/components/panel/UserActions.tsx]: Sure, i saw in different component usage without passing it to variable, so i can make this hook as variable and use in further functionality.

Comment (tlebon) [reply src/script/components/panel/UserActions.tsx]: maybe we should wait for @atomrc to weigh in on this also, but it seems **to me** like using the hook only is the correct action in this case.

Comment (thisisamir98) [reply src/script/components/panel/UserActions.tsx]: @przemvs you should destruct the variables you need and use them in the component, example:
ConnectedMessage.tsx

image

Comment (atomrc) [reply src/script/components/panel/UserActions.tsx]: @przemvs we might have a few places where the hook is used without reading the returned values, but this is mostly hacks/dirty fixes. We should really consume the returned values in the render.

In this case, that would mean replacing all occurrences of `user.<property>()` with the destructured property from the hook

Comment (przemvs) [reply src/script/components/panel/UserActions.tsx]: @atomrc  @tlebon @thisisamir98 fixed :)

**Review by atomrc — APPROVED** [review]
Great man. Awesome that you discovered the use of the `useKoSubscribableChildren`. 
Great job :+1:

File: src/script/components/panel/UserActions.tsx [hunk 2]
```diff
@@ -94,6 +94,35 @@ const UserActions: React.FC<UserActionsProps> = ({
   conversationRoleRepository,
   selfUser,
 }) => {
+  //isBlocked: false
+  // isCanceled: false
+  // isRequest: false
+  // isTeamMember: false
+  // isTemporaryGuest: false
+  // isUnknown: false
```
Comment (atomrc) [inline src/script/components/panel/UserActions.tsx]: Oh can we remove those comments

**Review by atomrc — CHANGES_REQUESTED** [review]

File: src/script/components/panel/UserActions.tsx (hunk 2 above)
Comment (atomrc) [inline src/script/components/panel/UserActions.tsx]:
```suggestion
```

**Review by atomrc — APPROVED** [review]

**Review by tlebon — APPROVED** [review]

**Review by thisisamir98 — APPROVED** [review]

**Pull request merged**